distributed/tests/test_nanny.py

```diff
@@ -484,10 +486,18 @@ async def start(self):
 
 @gen_cluster(nthreads=[])
 async def test_worker_start_exception(s):
-    # make sure this raises the right Exception:
-    with raises_with_cause(RuntimeError, None, StartException, None):
-        async with Nanny(s.address, worker_class=BrokenWorker) as n:
-            pass
+    nanny = Nanny(s.address, worker_class=BrokenWorker)
```

Comment:
1. We should change the `BrokenWorker.start` to `BrokenWorker.start_unsafe`. `Server.start` is no longer supposed to be overridden by subclasses after https://github.com/dask/distributed/pull/5910
2. Instead of capturing the logs, I believe we should do something like

```python
class BrokenWorker(worker.Worker):
    startup_attempts = 0
    async def start_unsafe(self):
        BrokenWorker.startup_attempts += 1
        raise StartException("broken")

def test_foo():
    BrokenWorker.startup_attempts = 0 # in case it is used elsewhere
    try:
        ...
        assert BrokenWorker.startup_attempts == 1
    finally:
        BrokenWorker.startup_attempts = 0 # in case it is used elsewhere

Reply:
Actually, it's not that easy, because the BrokenWorker is started in a subprocess, so the global variable doesn't carry over. We have to use a multiprocessing queue or shared memory.